app/routines/newflow/update_user_from_sheerid_webhook.rb

```diff
@@ -0,0 +1,58 @@
+module Newflow
+  class UpdateUserFromSheeridWebhook
+    lev_routine express_output: :user
+
+    protected ###############
+
+    def exec(verification_id:)
+      status.set_job_name(self.class.name)
+      status.set_job_args(verification_id: verification_id)
+
+      details = SheeridAPI.get_verification_details(verification_id)
+
+      last_response = details.fetch('lastResponse')
+      sheer_id_status = last_response.fetch('currentStep')
+      user_sheerid_info = details.fetch('personInfo')
+
+      email = user_sheerid_info.fetch('email')
+      outputs[:user] = user = User.find_by(sheerid_verification_id: verification_id)
+
+      fatal_error(code: :email_mismatch) if email_mismatch?(email, user)
+
+      user.update!(
+        first_name: user_sheerid_info.fetch('firstName'),
+        last_name: user_sheerid_info.fetch('lastName'),
+        sheerid_reported_school: user_sheerid_info.fetch('organization').fetch('name'),
+        faculty_status: (sheer_id_status == 'success' ? :confirmed_faculty : :pending_faculty),
+      )
+      create_security_log(user, details)
+    end
+
+    private #################
+
+    # true if no user found with such email or if that email email address belongs to someone else
+    def email_mismatch?(email, user)
+      existing_email = EmailAddress.find_by(value: email)
+
+      if !user.present? || !existing_email.present? || existing_email.user_id != user.id
```

Comment:
!user.present? might be a tiny bit clearer as user.blank?

Reply:
a code comment wouldnt have been necessary if this big IF statement was written as a method and then self-documenting

Reply:
True. Good point. I rewrote this part. Although, I'm not sure that I totally like the one-liner I turned it into 😅 . Will ask for re-review.